paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -108,6 +108,14 @@ void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
      */
     List<String> listTables(String databaseName) throws DatabaseNotExistException;
 
+    /**
+     * Check if a table exists in this catalog.
+     *
+     * @param identifier Path of the table
+     * @return true if the given table exists in the catalog false otherwise
+     */
+    boolean tableExists(Identifier identifier);
```

Comment:
I removed this method because it is very confused if we have system tables. You can find always return false for system tables in current implementations.